pytorch_lightning/callbacks/base.py

```diff
@@ -148,10 +148,22 @@ def on_predict_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningMo
         """Called when the predict epoch ends."""
 
     def on_epoch_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        """Called when either of train/val/test epoch begins."""
+        r"""
+        .. deprecated:: v1.6
+            This callback hook was deprecated in v1.6 and will be removed in v1.8. Use
+            ``on_<train/validation/test>_epoch_start`` instead.
+
+        Called when either of train/val/test epoch begins.
+        """
 
     def on_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        """Called when either of train/val/test epoch ends."""
+        r"""
+        .. deprecated:: v1.6
+            This callback hook was deprecated in v1.6 and will be removed in v1.8. Use
+            ``on_<train/validation/test>_epoch_end`` instead.
+
+        Called when either of train/val/test epoch ends.
```

Comment:
was it a bug that this wasn't called for predict? if so it underscores why its important to not duplicate hooks..

Reply:
Yes. I observed this back when I worked in the hook tests.

Removed its TODO in e6627bf

Reply:
looks like tests didn't like the removal of TODO comment 😄

Reply:
We are seeing the same failures in #11568, so seems it's not related to your PR

Reply:
ahaha..  I know.. a comment won't cause test failures. just joking around